.github/workflows/create-builds.yml

```diff
@@ -0,0 +1,112 @@
+# Summary:
+#
+# Reusable workflow for creating the distributable application assets.
+# Will also publish the application if the `app_type` input is `production`.
+#
+name: Create builds
+
+on:
+  workflow_call:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, release-candidate, or production.'
+        required: true
+        type: string
+      build_sha:
+        description: 'SHA of commit to build'
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, or release-candidate'
+        required: true
+        type: choice
+        # We only allow internal and release candidate builds when doing manual dispatches.
+        # Production builds should only be triggered by the appropriate calling workflows.
+        options: ['internal', 'release-candidate']
```

Comment:
I was thinking that allowing manual dispatch for these will be useful. The only question I have is whether production builds should also allowed to be manually dispatched. My intuition says no, but wondering how to handle cases where the production release build workflow fails for some reason. For example:

1. Merge RC PR -> create release tag
2. New release tag -> trigger production release
3. Production release fails -> ???

What's the course of action in (3)? Re-run the failed job? Maybe that's sufficient.